### ISSUE TITLE
Commit characters broken in simple cases

### ISSUE THREAD

**Issue by duzenko** [microsoft/vscode#181567]
# Bug Report

### 🔎 Search Terms

vscode complete dot
vscode commit characters
typescript commit characters

### 🕗 Version & Regression Information

Most likely never worked properly

### ⏯ Playground Link

[
[Playground link with relevant code](https://www.typescriptlang.org/play?#code/PTAEFkE9QYwewCYFNQHM5IM6gBZIE5JA)](https://www.typescriptlang.org/play?#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCbvCwDKgU8JkY7p7ehCTkVDQS2E6gnPCxGcwmZqDSTgzxxWWVoASMFmgYkAAeRJTInN3ymj4d-jSCeNsMq-wuoPaOltigAKoASgAywhK7SbGQZIIz5VWCFzSeCrZagNYbChbHaxUDcCjJZLfSDbExIAgUdxkUBIursJzCFJtXydajBBCcQQ0MwAUVWDEQC0gADVHBQGNJ3KAALygABEAAkYNAMOB4GRonzFBTBPB3AERcwABS0+mM9ysygc9wASkUAHEGMEVDhmh4yF5EOLnMD2DQvm4zMx2AxmJB+LBoOpEhz4HoRF4+XSGUzBHykti+eAHKgAOTCABykAkfKCKgA8l8yO9BO6tJAjqhHApXNI8KAnFadMMsPoUAhENDhJcHRJxdERAYONiFkxWBweEESkA)

### 💻 Code

```
const anExampleVariable = "Hello World"
console.log(anExampleVariable)
Ga<cursor>
```

### 🙁 Actual behavior

Typing `Ga`, then dot puts dot after Ga

### 🙂 Expected behavior

Whatever suggestion is selected (e.g. `GainNode`) must be completed and inserted in the code, followed by a dot

**Comment by duzenko** [comment]
@RyanCavanaugh Why did you moved the issue? Don't you know that similar issues have been auto-closed in the vscode tracker?

> Unfortunately we can't reliably compute commit characters on the VS Code extension side. This needs to come from the TypeScript server. Support for commit characters there is tracked by https://github.com/microsoft/TypeScript/issues/27623
https://github.com/microsoft/vscode/issues/110382#issuecomment-730702754

**Comment by mjbvz** [comment]
Works fine for me in VS Code. Is this only an issue in the TS playground?

**Comment by duzenko** [comment]
@mjbvz 
There are two snippets here: one fails in the playground, and the other is the real world example that fails in VSCode

**Comment by mjbvz** [comment]
Minimal example: 

```ts
function foo() {
    const x = fo // trigger suggest and type . here
}
```

@RyanCavanaugh I still think TS needs to be the one computing the commit characters. This will align it more with the LSP and let editors have a consistent UX

**Comment by VSCodeTriageBot** [comment]
This issue has been closed automatically because it needs more information and has not had recent activity. See also our [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.

Happy Coding!

**Comment by duzenko** [comment]
@mjbvz @RyanCavanaugh what info is needed exactly here?

**Comment by RyanCavanaugh** [comment]
@mjbvz is investigating what the next steps should be here in terms of whether TS server or VS code needs to change its behavior and how. We talked about this at the typescript vs code sync yesterday and everyone was in agreement that, ideally, dot is indeed a completion character here, we just need to work out the details of whether that's always the case and if not, how we figure out when it is.

**Comment by mjbvz** [comment]
@RyanCavanaugh  I investigated this further. We disable commit characters in this case because TypeScript returns `isNewIdentifierLocation` on the `completionInfo` response. This issue is actually already being tracked by https://github.com/microsoft/TypeScript/issues/31244

Closing this as a duplicate of https://github.com/microsoft/TypeScript/issues/31244. I still think it would make sense to have TS compute the commit characters instead of VS Code. For reference, here's where we compute these today:

https://github.com/microsoft/vscode/blob/5356e7ba59d49f27e78d5fc5285071eae927cb9f/extensions/typescript-language-features/src/languageFeatures/completions.ts#L495
